pkg/cmd/pulumi/up.go

```diff
@@ -877,3 +889,20 @@ func isPreconfiguredEmptyStack(
 
 	return true
 }
+
+func autonamingStackContext(s backend.Stack) autonaming.StackContext {
+	organization := "organization"
+	if cs, ok := s.(httpstate.Stack); ok {
+		organization = cs.OrgName()
+	}
+	project := "project"
+	if projName, ok := s.Ref().Project(); ok {
```

Comment:
At the point we call this we should have the Pulumi.yaml file available right? Maybe we should use that rather than s.Ref() to get the project name, then it will work for old DIY stacks as well.